### PR TITLE
Fix talker listener test

### LINKED ISSUE #83: talker_listener test is flaky
Following the same procedure as ros2/system_tests#93:

~~~
while true; do nosetests3 -s build/rclcpp_examples/test_talker_listener__rmw_opensplice_cpp.py; done
~~~

It eventually stops with this output:

~~~
(test_executable_0) pid 28945: ['/home/gerkey/ros2_ws/build/rclcpp_examples/talker__rmw_opensplice_cpp', 'test_executable'] (all > console, InMemoryHandler: test_executable_0)
(test_executable_1) pid 28946: ['/home/gerkey/ros2_ws/build/rclcpp_examples/listener__rmw_opensplice_cpp', 'test_executable'] (all > console, InMemoryHandler: test_executable_1)
[test_executable_0] Publishing: 'Hello World: 1'
[test_executable_0] Publishing: 'Hello World: 2'
(test_executable_0) rc -15
~~~

The `talker` program has now exited, leaving behind the `listener`, who still wants something. The `listener` backtrace is:

~~~
#0  pthread_cond_wait@@GLIBC_2.3.2 () at ../nptl/sysdeps/unix/sysv/linux/x86_64/pthread_cond_wait.S:185
#1  0x00007feb1a6f109b in RTIOsapiSemaphore_take () from /usr/lib/libnddscore.so.5
#2  0x00007feb1a9013c7 in PRESWaitSet_wait () from /usr/lib/libnddscore.so.5
#3  0x00007feb1bf53e6d in DDS_WaitSet_waitI () from /usr/lib/libnddsc.so.5
#4  0x00007feb1ba23ba3 in DDSWaitSet_impl::wait(DDSConditionSeq&, DDS_Duration_t const&) () from /usr/lib/libnddscpp.so.5
#5  0x00007feb1d9e3c28 in int wait<ConnextStaticSubscriberInfo, ConnextStaticServiceInfo, ConnextStaticClientInfo>(rmw_subscriptions_t*, rmw_guard_conditions_t*, rmw_services_t*, rmw_clients_t*, rmw_time_t*) ()
   from /home/gerkey/ros2_ws/install/lib/librmw_connext_cpp.so
#6  0x00007feb1d9e040d in rmw_wait () from /home/gerkey/ros2_ws/install/lib/librmw_connext_cpp.so
#7  0x00007feb1d6e9d30 in rclcpp::executor::Executor::wait_for_work(std::chrono::duration<long, std::ratio<1l, 1000000000l> >) ()
   from /home/gerkey/ros2_ws/install/lib/librclcpp.so
#8  0x00007feb1d6eaea0 in rclcpp::executor::Executor::get_next_executable(std::chrono::duration<long, std::ratio<1l, 1000000000l> >)
    () from /home/gerkey/ros2_ws/install/lib/librclcpp.so
#9  0x00007feb1d6f2cfb in rclcpp::executors::single_threaded_executor::SingleThreadedExecutor::spin() ()
   from /home/gerkey/ros2_ws/install/lib/librclcpp.so
#10 0x00007feb1d6ee22c in rclcpp::spin(std::shared_ptr<rclcpp::node::Node>) () from /home/gerkey/ros2_ws/install/lib/librclcpp.so
#11 0x000000000043d75f in main ()
~~~

### PR DESCRIPTION
**Pull request by jacquelinekay**
Fixes #83

I'm pull requesting this for @gerkey and queued a job that will repeat the test 20 times or until failure (simulating the conditions of the nightly job). http://ci.ros2.org/job/ci_osx/658

### CODE REVIEW

**Comment by dirk-thomas** [comment]
I would suggest making the test executable the last one since that's how it is in almost all ROS 1 rostest files.

File: rclcpp_examples/test/test_executables_example.py.in [hunk 1]
```diff
@@ -21,7 +21,10 @@ def test_executable():
     output_files = '@RCLCPP_EXAMPLES_EXPECTED_OUTPUT@'.split(';')
     for i, (exe, output_file) in enumerate(zip(executables, output_files)):
         name = 'test_executable_' + str(i)
-        handler = create_handler(name, launch_descriptor, output_file)
+        # The first executable is taken to be the test program (the one whose
+        # output check can make the decision to shut everything down)
+        test_process = True if i == 1 else False
```
Comment (gerkey) [inline rclcpp_examples/test/test_executables_example.py.in]: To support the case of N executables, I would go with:

~~~
test_process = True if i == (len(executables)-1) else False
~~~

Comment (dirk-thomas) [inline rclcpp_examples/test/test_executables_example.py.in]: For for simplicity of boolean logic that equates to:

    test_process = (i == (len(executables) - 1))

**Comment by dirk-thomas** [comment]
lgtm

**Pull request merged**